src/views/forms/AccountManagementForms/DepositForm.tsx

```diff
@@ -130,15 +130,19 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
   }, [debouncedAmountBN.toNumber()]);
 
   useEffect(() => {
-    abacusStateManager.setTransferValue({
-      field: TransferInputField.type,
-      value: TransferType.deposit.rawValue,
-    });
-
+    if (dydxAddress && evmAddress) {
+      // TODO: this is for fixing a race condition where the sourceAddress is not set in time.
+      // worth investigating a better fix on abacus
+      abacusStateManager.setTransfersSourceAddress(evmAddress);
+      abacusStateManager.setTransferValue({
+        field: TransferInputField.type,
+        value: TransferType.deposit.rawValue,
+      });
+    }
     return () => {
       abacusStateManager.resetInputState();
     };
-  }, []);
+  }, [dydxAddress]);
```

Comment:
should it also listen to evm address then?

Reply:
Sounds good, could add it to be safe! It is mainly that we want to trigger this once dydxAddress exists.